### PR TITLE
Refactor uniswap v3 changes

### PR DESCRIPTION
**Pull request by 2024river**
This PR refactors the uniswapv3 implementation.

### CODE REVIEW

**Review by shanshuo0918 — COMMENTED** [review]
1. We used to have a logic that automatically create new pools if the pool does not exist in the db. This is helpful when you start from a random block number.
2. Should use configuration to control which protocol to run.

**Comment by 2024river** [comment]
> 1. We used to have a logic that automatically create new pools if the pool does not exist in the db. This is helpful when you start from a random block number.
> 2. Should use configuration to control which protocol to run.

Initially, considering that the execution will start from the block where the pool is created, the step of calling RPC to determine whether to add a new pool is omitted.
However, this step can be added.

For configuration to run only a certain protocol.
In the pool job, the factory address can be used to fitler, but the amount of data is small and may not need to be configured.
For the price job, an additional pool address filter can be used, but the pools may change.
For the token job, it is currently controlled by the positions token.

**Comment by 2024river** [comment]
> 1. We used to have a logic that automatically create new pools if the pool does not exist in the db. This is helpful when you start from a random block number.
> 2. Should use configuration to control which protocol to run.

The logic of finding unsaved pools has been added.
Mainly when swap occurs and token positions change.
But it should be noted that although it is theoretically possible to start running from any block number.
However, the data that occurs between the pool creation and the block number selected by the program may not be available if it does not change later.

File: indexer/modules/custom/uniswap_v3/uniswap_v3_pool_price_job.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        self.multi_call_helper = MultiCallHelper(self._web3, kwargs, logger)
+        self.pools_requested_by_rpc = []
+
+    def get_filter(self):
+        address_list = self._pool_address if self._pool_address else []
+
+        return TransactionFilterByLogs(
+            [
+                TopicSpecification(
+                    topics=[
```
Comment (shanshuo0918) [inline indexer/modules/custom/uniswap_v3/uniswap_v3_pool_price_job.py:49]: 我觉得这个也应该配置控制？

另外配置要不改成类似数组的形式？这样你根据对应的type决定要filter那些事件？
type: uniswapv3
factory: 0xc848bc597903b4200b9427a3d7f61e3ff0553913
position_token_address: 0x7d24de60a68ae47be4e852cf03dd4d8588b489ec

type: agni
.....

Comment (2024river) [reply indexer/modules/custom/uniswap_v3/uniswap_v3_pool_price_job.py:49]: 嗯，配置格式改成这样合理吗：uniswap_v3_job:
  # empty means no filter, only work in pool job
  factory_address:

  # empty means no filter. only work in price job
  pool_address:

  # works in price\token job  
  jobs:
    - type: uniswapv3  # cle
      factory address: '0xaaa32926fce6be95ea2c51cb4fcb60836d320c42'
      position_token_address: '0xaaa78e8c4241990b4ce159e105da08129345946a'
    - type: swapsicle  # swapsicle
      factory address: '0xc848bc597903b4200b9427a3d7f61e3ff0553913'
      position_token_address: '0x7d24de60a68ae47be4e852cf03dd4d8588b489ec'
    - type: agni  # agni
      factory address: '0x25780dc8fc3cfbd75f33bfdab65e969b603b2035'
      position_token_address: '0x218bf598d1453383e2f4aa7b14ffb9bfb102d637'
    - type: agni  # fusion
      factory address: '0x530d2766d1988cc1c000c8b7d00334c14b69ad71'
      position_token_address: '0x5752f085206ab87d8a5ef6166779658add455774'

**Review by shanshuo0918 — APPROVED** [review]

**Pull request merged**